### PR TITLE
fix: [CDS-77549]: Input set GitX bug fixes

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
### Summary

Changes for:  https://github.com/harness/harness-core-ui/pull/17484

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sunnykesh-harness** [comment]
Prod deployment started for 0.358 so closing this hot fix for IFDs.

**Comment by sunnykesh-harness** [comment]
Reopening to merge this fix as suggested by @naidu-harness .

**Review by rameshx — APPROVED** [review]

File: src/modules/70-pipeline/components/InputSetForm/useSaveInputSet.ts [hunk 1]
```diff
@@ -112,6 +112,16 @@ export function useSaveInputSet(inputSetInfo: InputSetInfo): UseSaveInputSetRetu
   const { isGitSyncEnabled: isGitSyncEnabledForProject, gitSyncEnabledOnlyForFF } = React.useContext(AppStoreContext)
   const isGitSyncEnabled = isGitSyncEnabledForProject && !gitSyncEnabledOnlyForFF
 
+  const getinitialStoreMetadataPayload = React.useMemo(() => {
```
Comment (srdjan-harness) [inline src/modules/70-pipeline/components/InputSetForm/useSaveInputSet.ts:115]: getInitial....

Comment (srdjan-harness) [reply src/modules/70-pipeline/components/InputSetForm/useSaveInputSet.ts:115]: actually, its not a function we can remove "get"

**Pull request merged**